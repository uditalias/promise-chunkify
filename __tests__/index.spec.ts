import chunkify from "../index";

const resolver = (result: any, timeout: number = 0) => new Promise((resolve) => timeout ? setTimeout(() => resolve(result), timeout) : resolve(result));

describe('Promise Chunkify', () => {

    it('should run promises with concurrency', async () => {

        const results = await chunkify([
            () => resolver(1),
            () => resolver(2, 200),
            () => resolver(3),
        ], 1);

        expect(results).toMatchSnapshot();

    });
});