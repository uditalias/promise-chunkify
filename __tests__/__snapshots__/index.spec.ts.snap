// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promise Chunkify should run promises with concurrency 1`] = `
Array [
  1,
  2,
  3,
]
`;
